source/fem-dem/gls_vans.cc

```diff
@@ -644,120 +669,138 @@ GLSVANSSolver<dim>::quadrature_centered_sphere_method(bool load_balance_step)
               sum_quadrature_weights += fe_values_void_fraction.JxW(q);
             }
 
-          // if (!cell->at_boundary() || !cell->has_boundary_lines())
-          {
-            if (dim == 2)
-              R_sphere = (std::sqrt(cell->measure() / M_PI));
-            else if (dim == 3)
-              R_sphere = (pow(3 * cell->measure() / (4 * M_PI), 1. / 3.));
-
-            // Array of real locations for the quadrature points
-            std::vector<Point<dim>> quadrature_point_location;
+          double reference_sphere_volume;
 
-            quadrature_point_location =
-              fe_values_void_fraction.get_quadrature_points();
+          if (this->cfd_dem_simulation_parameters.void_fraction
```

Comment:
I think here I would do the opposite. If sphere_element_equal_volume it does not use reference_sphere_diameter. Then if it was false, you would use reference_sphere_diameter. In my opinion, it would be cleaner and would not depend on a double

Reply:
In this case, i don't believe so. Because if the dp = 0 then we do it automatically, if dp = value then it uses this values. If dp =0, we have two options: if boolean is true, we use the same value for the sphere as the cell, if it is false, then we use radius of sphere size of the cell.

Reply:
I like Toni approach, but plaese do not use "if conditions" with double. use seomthing like < 1e-16 or somethign similar to prevent weird issues.